lib/WUI/link_content/basic_gets.cpp

```diff
@@ -0,0 +1,223 @@
+    // clang-format off
+    JSON_START;
+    JSON_OBJ_START;
+        JSON_FIELD_OBJ("telemetry");
+            JSON_FIELD_FFIXED("temp-bed", vars->temp_bed) JSON_COMMA;
+            JSON_FIELD_FFIXED("temp-nozzle", vars->temp_nozzle) JSON_COMMA;
+            JSON_FIELD_INT("print-speed", vars->print_speed) JSON_COMMA;
+            // XYZE, mm
+            JSON_FIELD_FFIXED("z-height", vars->pos[2]) JSON_COMMA;
+            JSON_FIELD_STR("material", filament_material);
+        JSON_OBJ_END JSON_COMMA;
+
+        JSON_FIELD_OBJ("temperature");
+            JSON_FIELD_OBJ("tool0");
+                JSON_FIELD_FFIXED("actual", vars->temp_nozzle) JSON_COMMA;
+                JSON_FIELD_FFIXED("target", vars->target_nozzle) JSON_COMMA;
+                // Note: our own extension, because our printers sometimes display
+                // different "target" temperature than what they heat towards.
+                JSON_FIELD_FFIXED("display", vars->display_nozzle) JSON_COMMA;
+                JSON_FIELD_INT("offset", 0);
+            JSON_OBJ_END JSON_COMMA;
+            JSON_FIELD_OBJ("bed");
+                JSON_FIELD_FFIXED("actual", vars->temp_bed) JSON_COMMA;
+                JSON_FIELD_FFIXED("target", vars->target_bed) JSON_COMMA;
+                JSON_FIELD_INT("offset", 0);
+            JSON_OBJ_END;
+        JSON_OBJ_END JSON_COMMA;
+
+        JSON_FIELD_OBJ("state")
+            JSON_FIELD_STR("text", printing ? "Printing" : "Operational") JSON_COMMA;
+            JSON_FIELD_OBJ("flags");
+                JSON_FIELD_BOOL("operational", operational) JSON_COMMA;
+                JSON_FIELD_BOOL("paused", paused) JSON_COMMA;
+                JSON_FIELD_BOOL("printing", printing) JSON_COMMA;
+                JSON_FIELD_BOOL("cancelling", cancelling) JSON_COMMA;
+                JSON_FIELD_BOOL("pausing", pausing) JSON_COMMA;
+                // We don't have an SD card!
+                JSON_CONTROL("\"sdReady\":false,\"error\":false,\"closedOnError\":false,");
+                JSON_FIELD_BOOL("ready", ready) JSON_COMMA;
+                JSON_FIELD_BOOL("busy", busy);
+            JSON_OBJ_END;
+        JSON_OBJ_END;
+    JSON_OBJ_END;
+    JSON_END;
+    // clang-format off
+}
+
+JsonRenderer::ContentResult get_version(size_t resume_point, JsonRenderer::Output &output) {
+    /*
+     * FIXME: The netdev_get_hostname doesn't properly synchronize. That needs
+     * a fix of its own. But to not make things even worse than they are, we
+     * make sure to copy it out to our side first and make sure it doesn't
+     * change during the JSON stringification which could lead to a different
+     * length of the output and stack smashing.
+     */
+    const char *hostname_unsynchronized = netdev_get_hostname(netdev_get_active_id());
+    const size_t hostname_in_len = strlen(hostname_unsynchronized);
+    char hostname[hostname_in_len + 1];
+    memcpy(hostname, hostname_unsynchronized, hostname_in_len);
```

Comment:
Might include size of `char`.

Reply:
Again, not a new code. Furthermore, `sizeof char` is by definition `1`. That's how sizeof itself is defined, by number of chars.

Reply:
I meant `typeof(*hostname)` which is `char` which is `1`. But let's close this.